src/tests/ftest/datamover/dm_posix_symlinks.py

```diff
@@ -0,0 +1,256 @@
+        self.start_dfuse(self.dfuse_hosts)
+
+        # Create 1 pool
+        pool1 = self.create_pool()
+
+        # Create a special container to hold UNS entries
+        uns_cont = self.create_cont(pool1)
+
+        # Test links that point forward
+        container1 = self.create_cont(pool1, True, pool1, uns_cont)
+        self.run_dm_posix_symlinks_fun(
+            pool1, container1, self.create_links_forward, "forward")
+
+        # Test links that point backward
+        container2 = self.create_cont(pool1, True, pool1, uns_cont)
+        self.run_dm_posix_symlinks_fun(
+            pool1, container2, self.create_links_backward, "backward")
+
+        # Test a mix of forward and backward links
+        container3 = self.create_cont(pool1, True, pool1, uns_cont)
+        self.run_dm_posix_symlinks_fun(
+            pool1, container3, self.create_links_mixed, "mixed")
+
+    def run_dm_posix_symlinks_fun(self, pool, cont, link_fun, link_desc):
+        """
+        Tests copying symlinks with and without --dereference.
+
+        Args:
+            pool (TestPool): The pool to use
+            cont (TestContainer): The container for both src and dst
+            link_fun (str -> void): The function for creating the
+                symlink structure. A path is passed for the location.
+            link_desc (str): A description about the link_fun.
+                Used in logging.
+        """
+        # Get the dereference param
+        do_deref = self.params.get(
+            "dereference", "/run/{}/*".format(self.tool.lower()))
+
+        # Use a common test_desc
+        test_desc = self.test_id + "({})".format(link_desc)
+        test_desc += " (dereference={})".format(str(do_deref))
+        self.log.info("Running %s", test_desc)
+
+        # Get a directory for POSIX
+        posix_test_path = self.new_posix_test_path()
+
+        # Save some paths and encode the type in the path for easier debugging
+        src_daos_dir = "/src_" + link_desc
+        src_daos_path = cont.path.value + src_daos_dir
+        src_posix_path = join(posix_test_path, "src_" + link_desc)
+
+        # Create the source links
+        link_fun(src_daos_path)
+        link_fun(src_posix_path)
+
+        if do_deref:
+            # Use POSIX cp to create a baseline for dereferencing
+            deref_baseline_path = join(posix_test_path, "baseline_" + link_desc)
+            self.execute_cmd("cp -r --dereference '{}' '{}'".format(
+                src_posix_path, deref_baseline_path))
+            diff_src = deref_baseline_path
+        else:
+            # Just compare against the original
+            diff_src = src_posix_path
+
+        # DAOS -> DAOS
+        dst_daos_dir = self.new_daos_test_path(False)
+        self.run_datamover(
+            test_desc + " (DAOS->DAOS)",
+            "DAOS", src_daos_dir, pool, cont,
+            "DAOS", dst_daos_dir, pool, cont)
+        self.run_diff(diff_src, cont.path.value + dst_daos_dir, do_deref)
```

Comment:
I believe you're calling diff to verify that the copying has been done successfully. How are you checking the output of the diff command?

Reply:
`diff` returns 0 if there are no differences found, and 1 if there are differences found. In the case that it returns 1, `self.execute_cmd` will call `self.fail`. If needed, we could be more explicit and pass `--report-identical-files` and check the diff output for "Files <src> and <dst> are identical"